### PR TITLE
Refactors and fixes Kanada code.

### PR DESCRIPTION
**Pull request by GuyonBroadway**
:cl: Guyonbroadway
fix: Fixed the action buttons on the Kanada set, you can now use nice UI icons to produce pucks and get your hockey stick.
refactor: Replaced shitty kanada verbs with procs.
/:cl:

Why?  https://youtu.be/u9vxsTaWu1s

Also Amari Ray broke most of Kanada code because of his hamfisted "fixes"

### CODE REVIEW

**Comment by Manly-Dwarf** [comment]
Amari doesn't test his hamfisted nerfs/shitcode?

Damn, couldn't have seen that one coming.

thanks for this by the way

File: hippiestation/code/modules/sports/hockey.dm [hunk 1]
```diff
@@ -187,13 +199,18 @@
 	var/obj/item/holopuck/newpuck
 
 /obj/item/storage/belt/hippie/hockey/equipped(mob/user, slot)
+	..()
 	if(slot == slot_belt)
 		flags_1 |= NODROP_1
 
+/obj/item/storage/belt/hippie/hockey/item_action_slot_check(slot, mob/user)
+	if(slot == user.getBeltSlot())
+		return 1
```
Comment (JohnGinnane) [inline hippiestation/code/modules/sports/hockey.dm]: `return TRUE`

File: hippiestation/code/modules/sports/hockey.dm [hunk 2]
```diff
@@ -92,6 +96,10 @@
 		return
 	..()
 
+/obj/item/hockeypack/item_action_slot_check(slot, mob/user)
+	if(slot == user.getBackSlot())
+		return 1
```
Comment (GuyonBroadway) [inline hippiestation/code/modules/sports/hockey.dm]: waaay aheads of you on this one

Comment (JohnGinnane) [inline hippiestation/code/modules/sports/hockey.dm]: idk if you saw but same here also pls

File: hippiestation/code/modules/sports/hockey.dm [hunk 3]
```diff
@@ -54,8 +57,9 @@
 /obj/item/hockeypack/proc/make_stick()
 	return new /obj/item/twohanded/hockeystick(src)
 
-/obj/item/hockeypack/equipped(mob/user, slot) //The Pack is cursed so this should not happen, but i'm going to play it safe.
-	if (slot != slot_back)
+/obj/item/hockeypack/equipped(mob/user, slot)
```
Comment (JohnGinnane) [inline hippiestation/code/modules/sports/hockey.dm]: I think we should just remove this duplicate proc altogether

File: hippiestation/code/modules/sports/hockey.dm [hunk 4]
```diff
@@ -123,11 +131,11 @@
 	icon_state = "hockeystick[wielded]"
 	return
 
-/obj/item/twohanded/hockeystick/Initialize(parent_pack)
-	. = ..()
+/obj/item/twohanded/hockeystick/New(parent_pack)
```
Comment (JohnGinnane) [inline hippiestation/code/modules/sports/hockey.dm:129]: https://www.youtube.com/watch?v=oHC1230OpOg

why New() over Init()

File: hippiestation/code/modules/sports/hockey.dm (hunk 3 above)
Comment (GuyonBroadway) [reply hippiestation/code/modules/sports/hockey.dm]: fuck i left that in...

File: hippiestation/code/modules/sports/hockey.dm (hunk 4 above)
Comment (GuyonBroadway) [reply hippiestation/code/modules/sports/hockey.dm:129]: I was basing it off defibrilator code. They use new for whatever reason, i could change it back but so far it works.

Comment (JohnGinnane) [reply hippiestation/code/modules/sports/hockey.dm:129]: I suspect because there won't be any on map loading so yeah this maps sense

**Review by JohnGinnane — APPROVED** [review]

File: hippiestation/code/modules/sports/hockey.dm (hunk 4 above)
Comment (GuyonBroadway) [reply hippiestation/code/modules/sports/hockey.dm:129]: which is funny because Defibs use it and there ARE defibs on map load...

Comment (JohnGinnane) [reply hippiestation/code/modules/sports/hockey.dm:129]: Defibs yeah, but not their paddles

**Comment by JohnGinnane** [comment]
Just gonna try and sort the tgui thing before this if that's okay

**Pull request merged**